data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/PstRangeActionAdderImpl.java

```diff
@@ -0,0 +1,116 @@
+import com.farao_community.farao.data.crac_impl.usage_rule.FreeToUse;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static com.farao_community.farao.data.crac_api.UsageMethod.AVAILABLE;
+
+/**
+ * @author Peter Mitri {@literal <peter.mitri at rte-france.com>}
+ */
+public class PstRangeActionAdderImpl extends AbstractIdentifiableAdder<PstRangeActionAdderImpl> implements PstRangeActionAdder {
+    private SimpleCrac parent;
+    private Unit unit;
+    private Double minValue;
+    private Double maxValue;
+    private NetworkElement networkElement;
+    private String operator;
+    private List<UsageRule> usageRules;
+
+    public PstRangeActionAdderImpl(SimpleCrac parent) {
+        Objects.requireNonNull(parent);
+        this.parent = parent;
+        this.usageRules = new ArrayList<>();
+        this.usageRules.add(new FreeToUse(AVAILABLE, parent.getPreventiveState()));
+    }
+
+    @Override
+    public PstRangeActionAdder setOperator(String operator) {
+        this.operator = operator;
+        return this;
+    }
+
+    @Override
+    public PstRangeActionAdder setUnit(Unit unit) {
+        this.unit = unit;
+        return this;
+    }
+
+    @Override
+    public PstRangeActionAdder setMinValue(Double minValue) {
+        this.minValue = minValue;
+        return this;
+    }
+
+    @Override
+    public PstRangeActionAdder setMaxValue(Double maxValue) {
+        this.maxValue = maxValue;
+        return this;
+    }
+
+    @Override
+    public NetworkElement addNetworkElement(NetworkElement networkElement) {
+        this.networkElement = networkElement;
+        return networkElement;
+    }
+
+    @Override
+    public NetworkElementAdder newNetworkElement() {
+        if (networkElement == null) {
+            return new NetworkElementAdderImpl<PstRangeActionAdder>(this);
+        } else {
+            throw new FaraoException("You can only add one network element to a PstRangeAction.");
+        }
+    }
+
+    @Override
+    public Crac add() {
+        checkId();
+        if (this.unit == null) {
+            throw new FaraoException("Cannot add a PstRangeAction without a unit. Please use setUnit.");
+        }
+        if (this.unit != Unit.TAP) {
+            throw new FaraoException("Only TAP unit is currently supported.");
+        }
+        if (this.minValue == null) {
+            throw new FaraoException("Cannot add a PstRangeAction without a minimum value. Please use setMinValue.");
+        }
+        if (this.maxValue == null) {
+            throw new FaraoException("Cannot add a PstRangeAction without a maximum value. Please use setMaxValue.");
+        }
+        if (this.networkElement == null) {
+            throw new FaraoException("Cannot add a PstRangeAction without a network element. Please use newNetworkElement.");
+        }
+        List<Range> ranges = new ArrayList<>();
```

Comment:
`List<Range> ranges = Collections.singletonList(new Range(this.minValue, this.maxValue, RangeType.ABSOLUTE_FIXED, RangeDefinition.CENTERED_ON_ZERO));`